DependencyInjection/FOSUserExtension.php

```diff
@@ -133,6 +132,22 @@ public function load(array $configs, ContainerBuilder $container)
         if (!empty($config['group'])) {
             $this->loadGroups($config['group'], $container, $loader, $config['db_driver']);
         }
+
+        if ($container->hasParameter('fos_user.registration.confirmation.from_email') || $container->hasParameter('fos_user.resetting.email.from_email')) {
+            $mailer = $config['service']['mailer'];
+            $container->setAlias('fos_user.mailer', $mailer);
+            $mailerDefinition = $container->getDefinition($mailer);
```

Comment:
The mailer should then be configured through a CompilerPass. However, you will still need a way to provide parameters. If that's pretty straightforward using one of the default provided mailer services, it could become more tricky with a custom one. Then again, I can't find any better solution than using an interface. Maybe you would have a better idea on how to deal with it ?